tests/unit_tests/ode_solver_unit_test/explicit_ode_solver.cpp

```diff
@@ -0,0 +1,220 @@
+
+    const double adv_speed_x = 1.0, adv_speed_y = 0.0;
+    all_parameters.manufactured_convergence_study_param.manufactured_solution_param.advection_vector[0] = adv_speed_x; //x-velocity
+    all_parameters.manufactured_convergence_study_param.manufactured_solution_param.advection_vector[1] = adv_speed_y; //y-velocity
+
+    int n_time_calculations = 4;
+    const double dt_init = 2.5E-3;
+    double dt = dt_init;
+    const double refine_ratio = 0.5;
+    const double finalTime = 1.0;
+
+    dealii::ConvergenceTable convergence_table;
+
+    double L2_error_old = 0;
+    dealii::Vector<double> L2_error_conv_rate(n_time_calculations);
+
+    for (int refinement = 0; refinement < n_time_calculations; ++refinement){
+
+        //initial_time_step is not modified by explicit ODE solver
+        all_parameters.ode_solver_param.initial_time_step = dt;
+        std::cout << "Using time step = " << dt << std::endl;
+        std::cout << "refinement = " << refinement << std::endl;
+
+        all_parameters.ode_solver_param.output_solution_every_x_steps = int(finalTime/dt/10.0); //output 10 vtk files (if dt reaches finalTime exactly)
+
+        const unsigned int space_poly_degree = rk_order + 2;
+        std::shared_ptr < PHiLiP::DGBase<dim, double> > dg = PHiLiP::DGFactory<dim,double>::create_discontinuous_galerkin(&all_parameters, space_poly_degree, grid);
+        dg->allocate_system ();
+
+        //initial conditions
+        std::cout << "Implement initial conditions" << std::endl;
+        dealii::FunctionParser<dim> initial_condition;
+        std::string variables;
+        std::map<std::string,double> constants;
+        constants["pi"] = dealii::numbers::PI;
+        std::string expression_initial;
+        if (dim == 2){
+            variables = "x,y";
+            expression_initial = "exp( -( 20*(x-1)*(x-1) + 20*(y-1)*(y-1) ) )";
+        }
+        else if (dim == 1){
+            variables = "x";
+            //expression_initial = "exp(- 20 * (x-1) * (x-1))";	
+            expression_initial = "sin(2*pi*x/2.0)";	
+        }
+        initial_condition.initialize(variables,
+                expression_initial,
+                constants);
+        dealii::VectorTools::interpolate(dg->dof_handler,initial_condition,dg->solution);
+
+        // Create ODE solver using the factory and providing the DG object
+        std::shared_ptr<PHiLiP::ODE::ODESolverBase<dim, double>> ode_solver = PHiLiP::ODE::ODESolverFactory<dim, double>::create_ODESolver(dg);
+
+        //Solve
+        ode_solver->advance_solution_time(finalTime);
+
+        //Defining exact solution
+        dealii::FunctionParser<dim> exact_solution;
+        constants["a_x"] = adv_speed_x;
+        constants["a_y"] = adv_speed_y;
+        constants["t"] = ode_solver -> current_time;//finalTime;
+        std::string expression_exact;
+        if (dim == 2){
+            expression_exact = "exp( -( 20*(x-1-a_x*t)*(x-1-a_x*t) + 20*(y-1-a_y*t)*(y-1-a_y*t) ) )";
```

Comment:
If these are the exact solutions to the unsteady problem you're simulating and checking the order with, maybe we should have some sort of `ExactSolutionFactory` similar to `InitialConditionFactory`, where you'd create the directory `src/physics/exact_solutions` and create the files in a similar fashion -- this will be good for when we want to check the temporal order of convergence for different problems